inbox/mailsync/backends/base.py

```diff
@@ -22,6 +26,8 @@ class MailsyncDone(InterruptibleThreadExit):
 
 
 class BaseMailSyncMonitor(InterruptibleThread):
+    sync_engine_class: ClassVar[type["FolderSyncEngine"]]
+
```

Comment:
I've moved this out of an initializer so the thread class used to sync a single folder can be accessed without constructing a monitor. A monitor is a parent thread used to sync all the folders.